.azure-pipelines/ultimate-pipeline.yml

```diff
@@ -1490,6 +1487,65 @@ stages:
       condition: succeededOrFailed()
       continueOnError: true
 
+- stage: static_analysis_checks_tracer
+  dependsOn: [merge_commit_id]
+  variables:
+    targetShaId: $[ stageDependencies.merge_commit_id.fetch.outputs['set_sha.sha']]
+    targetBranch: $[ stageDependencies.merge_commit_id.fetch.outputs['set_sha.branch']]
+  jobs:
+  - template: steps/update-github-status-jobs.yml
+    parameters:
+      jobs: [linux, macos]
+
+  - job: linux
+    timeoutInMinutes: 60 #default value
+    dependsOn: []
+    strategy:
+      matrix:
+        centos7:
+          baseImage: centos7
+          poolName: azure-linux-scale-set
+        alpine:
+          baseImage: alpine
+          poolName: azure-linux-scale-set
+        arm64:
+          baseImage: debian
+          poolName: aws-arm64-auto-scaling
```

Comment:
wouldn't it be simpler to run cppcheck only on cento7 ?

Reply:
> wouldn't it be simpler to run cppcheck only on centos7

Yes, definitely simpler, this is just what we had before, so thought it was necessary - if not, I'll simplify it, thanks!

Reply:
done in https://github.com/DataDog/dd-trace-dotnet/pull/4690/commits/24092e999e149f710f65afc66a193cec71718570